File: software/server/irrigation/domain/models/actuator.py
# -*- coding: utf-8 -*-
import attr


@attr.s
class ActuatorData():
    """ Data stored by actuator
    """
    area_id: int = attr.ib()
    area: str = attr.ib()
    start_date: str = attr.ib(default=None)
    end_date: str = attr.ib(default=None)


@attr.s
class QueryActuatorData():
    area_id: int = attr.ib()
    start_date: str = attr.ib(default=None)
    end_date: str = attr.ib(default=None)

File: software/web/src/api/services/actuator.ts
import { Actuator } from '../../types/actuator';
import { axios } from '../axios';
import { API_ACTUATOR_ACTIVATION_HISTORICAL } from '../urls';


interface ResponseActuatorActivationHistorical {
    data: {
        actuator_activation: Record<string, string>[];
    };
    status: string,
}

export const getHistorical = async (area: number, startDate: string, endDate: string): Promise<Actuator[]> => {
    const queryParameters = `?start_date=${startDate}&end_date=${endDate}`;

    return axios.get<any, ResponseActuatorActivationHistorical>(`${API_ACTUATOR_ACTIVATION_HISTORICAL.replace(':area', area.toString())}${queryParameters}`).then((data) => {
        if (data) {
            return data.data.actuator_activation.map(i => {
                return {
                    "area": i.area,
                    "start_date": i.start_date,
                    "end_date": i.end_date,
                } as unknown as Actuator
            });
        }

        return Promise.reject();
    })
}
